### ISSUE TITLE
CoreDump Occurred in Process Running Within Nvidia PyTorch Runtime Image

### ISSUE THREAD

**Issue by lvdunlin** [facebookincubator/dynolog#353]
Recently tested using dynolog to dynamically collect profile data for PyTorch programs. It was found that CoreDump occurs in the following Nvidia PyTorch runtime images, specifically as follows:

nvcr.io/nvidia/pytorch:25.01-py3 

![Image](https://github.com/user-attachments/assets/e04ba395-ca30-482d-9cfa-a7ac390c4122)

nvcr.io/nvidia/pytorch:25.02-py3 

![Image](https://github.com/user-attachments/assets/1210982a-be96-4951-8ea7-257fa0e48d24)

GPU: A800

My test program is as follows:
import os
import torch
import torch.nn as nn
import time
from torchvision.models import resnet18

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
model = resnet18().to(device)
inputs = torch.randn(800, 3, 224, 224).to(device)
targets = torch.randint(0, 1000, (800,)).to(device)
criterion = nn.CrossEntropyLoss()
optimizer = torch.optim.SGD(model.parameters(), lr=0.001)

def test(iterations=200):
    print("pid: {}".format(os.getpid()))
    _ = model(inputs)
    
    times = []
    for idx in range(1, iterations):
        start_time = time.perf_counter()
        
        outputs = model(inputs)
        loss = criterion(outputs, targets)
        
        optimizer.zero_grad()
        loss.backward()
        optimizer.step()
        
        times.append(time.perf_counter() - start_time)
        print("Iter {} cost {:.3f}".format(idx, times[-1]))

    return times

if __name__ == "__main__":
    test()

**Comment by lvdunlin** [comment]
@briancoutinho

**Comment by briancoutinho** [comment]
@lvdunlin sorry about the delay this seems like kineto issue to me. Can you share what is the PyTorch version you see this on?

**Comment by lvdunlin** [comment]
> [@lvdunlin](https://github.com/lvdunlin) sorry about the delay this seems like kineto issue to me. Can you share what is the PyTorch version you see this on?

@briancoutinho  docker image: nvcr.io/nvidia/pytorch:25.01-py3 pytorch: 2.6.0a0+ecf3bae40a.nv25.1
